src/healthkit.cs

```diff
@@ -2442,4 +2557,220 @@ interface HKWorkoutRouteQuery {
+
+		[Async]
+		[Export ("beginCollectionWithStartDate:completion:")]
+		void BeginCollection (NSDate startDate, Action<bool, NSError> completionHandler);
+
+		[Async]
+		[Export ("addSamples:completion:")]
+		void Add (HKSample[] samples, Action<bool, NSError> completionHandler);
+
+		[Async]
+		[Export ("addWorkoutEvents:completion:")]
+		void Add (HKWorkoutEvent[] workoutEvents, Action<bool, NSError> completionHandler);
+
+		[Async, Internal]
+		[Export ("addMetadata:completion:")]
+		void Add (NSDictionary metadata, Action<bool, NSError> completionHandler);
+
+		[Async]
+		[Wrap ("Add (metadata.Dictionary, completionHandler)")]
+		void Add (HKMetadata metadata, Action<bool, NSError> completionHandler);
+
+		[Async]
+		[Export ("endCollectionWithEndDate:completion:")]
+		void EndCollection (NSDate endDate, Action<bool, NSError> completionHandler);
+
+		[Async]
+		[Export ("finishWorkoutWithCompletion:")]
+		void FinishWorkout (Action<HKWorkout, NSError> completionHandler);
+
+		[Export ("discardWorkout")]
+		void DiscardWorkout ();
+
+		[Export ("elapsedTimeAtDate:")]
+		double ElapsedTimeAtDate (NSDate date);
+
+		[Export ("statisticsForType:")]
+		[return: NullAllowed]
+		HKStatistics Statistics (HKQuantityType quantityType);
+
+		[Export ("seriesBuilderForType:")]
+		HKSeriesBuilder SeriesBuilder (HKSeriesType seriesType);
+	}
+
+	delegate void HKQuantitySeriesSampleQueryQuantityDelegate (HKQuantitySeriesSampleQuery query, HKQuantity quantity, NSDate date, bool done, NSError error);
+
+	[Watch (5,0), iOS (12,0)]
+	[BaseType (typeof(HKQuery))]
+	interface HKQuantitySeriesSampleQuery
+	{
+		[Export ("initWithSample:quantityHandler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (HKQuantitySample quantitySample, HKQuantitySeriesSampleQueryQuantityDelegate quantityHandler);
+	}
+
+	[Watch (5,0), iOS (12,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface HKQuantitySeriesSampleBuilder
+	{
+		[Export ("initWithHealthStore:quantityType:startDate:device:")]
+		IntPtr Constructor (HKHealthStore healthStore, HKQuantityType quantityType, NSDate startDate, [NullAllowed] HKDevice device);
+
+		[Export ("quantityType", ArgumentSemantic.Copy)]
+		HKQuantityType QuantityType { get; }
+
+		[Export ("startDate", ArgumentSemantic.Copy)]
+		NSDate StartDate { get; }
+
+		[NullAllowed, Export ("device", ArgumentSemantic.Copy)]
+		HKDevice Device { get; }
+
+		[Export ("insertQuantity:date:error:")]
+		bool Insert (HKQuantity quantity, NSDate date, [NullAllowed] out NSError error);
+
+		[Async, Internal]
+		[Export ("finishSeriesWithMetadata:completion:")]
+		void FinishSeries ([NullAllowed] NSDictionary metadata, Action<NSArray<HKQuantitySample>, NSError> completionHandler);
+
+		[Async]
+		[Wrap ("FinishSeries (metadata.Dictionary, completionHandler)")]
+		void FinishSeries ([NullAllowed] HKMetadata metadata, Action<NSArray<HKQuantitySample>, NSError> completionHandler);
```

Comment:
same `NSArray<HKQuantitySample>` vs `HKQuantitySample []` I do not recall if Action supports the managed array version, if not just use a named delegate